src/Specular/Dom/Element.purs

```diff
@@ -0,0 +1,61 @@
+module Specular.Dom.Element
+  ( module Specular.Dom.Element
+  , module X
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Maybe (Maybe(..))
+import Data.Tuple (Tuple(..), snd)
+import Effect (foreachE)
+import Effect.Uncurried (EffectFn2, mkEffectFn1, mkEffectFn2, runEffectFn1, runEffectFn2)
+import Foreign.Object as Object
+import Specular.Dom.Browser (Node, appendChild)
+import Specular.Dom.Builder.Class (liftBuilderWithRun)
+import Specular.Dom.Builder.Class (text, rawHtml) as X
+import Specular.Dom.Node.Class (Attrs, TagName, createElement, removeAttributes, setAttributes)
+import Specular.Dom.Widget (class MonadWidget)
+import Specular.FRP (Dynamic, readDynamic, _subscribeEvent, changed)
+import Specular.Internal.Effect (DelayedEffects, newRef, readRef, writeRef, pushDelayed)
+
+newtype Prop = Prop (EffectFn2 Node DelayedEffects Unit)
+
+el' :: forall m a. MonadWidget m => TagName -> Array Prop -> m a -> m (Tuple Node a)
+el' tagName props body = liftBuilderWithRun $ mkEffectFn2 \env run -> do
+  node <- createElement tagName
+  result <- runEffectFn2 run (env { parent = node }) body
+  foreachE props \(Prop prop) ->
+    runEffectFn2 prop node env.cleanup
+  appendChild node env.parent
+  pure (Tuple node result)
+
+el :: forall m a. MonadWidget m => TagName -> Array Prop -> m a -> m a
```

Comment:
Maybe we should have also `el' :: TagName -> m a -> m a` (no props) for consistency? That's quite common case.

Reply:
Neat idea, but requires a bit of bikeshedding. We're running out of suffixes here:

- `el` is the main version
- `el'` is the version which also returns the node
- `el_` is a shortcut for elements without children

Note: In Halogen, for example, the `_` suffix is for the tweak you're proposing - omitting props.